### ISSUE TITLE
The async task did not return valid JSON    RHEL5 only

### ISSUE THREAD

**Issue by binbashroot** [ansible/ansible#15162]
##### ISSUE TYPE

- Bug Report
##### ANSIBLE VERSION

```
“ansible 2.0.1.0
  config file = /home/binbashroot/ansible-prod.cfg
  configured module search path = /usr/lib/python2.6/site-packages/ansible/modules
” 

```
##### OS / ENVIRONMENT

RHEL5/6
##### SUMMARY

Tasks using async/poll with RH5 servers fails 
##### STEPS TO REPRODUCE

```
<!--- Paste example playbooks or commands between quotes -->
```
## "
- hosts: alpha
  gather_facts: True
  become: True
  tasks:
  - debug: msg="{{ansible_distribution_version}}"
    when: ansible_os_family == "RedHat"
  - name: Test Async with rhel5 and rhel6
    shell: /bin/sleep 30
    async: 45
    poll: 7
    when: ansible_os_family == "RedHat"
    "
    
    
##### ACTUAL RESULTS

"
PLAY ***************************************************************************

TASK [setup] *******************************************************************
ok: [p1rhrep]
ok: [dgitapp01]

TASK [debug] *******************************************************************
ok: [p1rhrep] => {
    "msg": "5.11"
}
ok: [dgitapp01] => {
    "msg": "6.7"
}

TASK [Test Async with rhel5 and rhel6] *****************************************
fatal: [p1rhrep]: FAILED! => {"changed": false, "failed": true, "msg": "The async task did not return valid JSON: Extra data: line 2 column 1 - line 5 column 1 (char 115 - 244)"}
changed: [dgitapp01]

NO MORE HOSTS LEFT *************************************************************

PLAY RECAP *********************************************************************
dgitapp01                  : ok=3    changed=1    unreachable=0    failed=0
p1rhrep                    : ok=2    changed=0    unreachable=0    failed=1

"

```
<!--- Paste verbatim command output between quotes -->
```

**Comment by jonathannaguin** [comment]
I have had a similar issue: host is RHEL5 and Ansible 2.0.2.0 complained about the JSON returned by Async.
I fixed upgrading my Python in the RHEL5 to Python 2.6, which contains bundled the JSON module.

**Comment by mattclay** [comment]
PR #16534 been merged and may fix your issue. The problem may go away entirely with this fix. If not, the error message "The async task did not return valid JSON" you reported should be replaced with a more informative error, allowing further troubleshooting.

Unless further tests reveal major bugs, this fix will be included in the upcoming 2.1.1 release. If you'd like to test and confirm the fix before then, please follow the instructions on [running from source](http://docs.ansible.com/ansible/intro_installation.html#running-from-source). Be sure to use `-b stable-2.1` when you clone so you are testing the stable-2.1 branch.

If we don't hear back before the 2.1.1 release, we'll consider this issue resolved and close the issue. If the problem is not resolved, please update this issue with the relevant details.

**Comment by mattclay** [comment]
Closing as resolved with the 2.1.1 release.
